### PR TITLE
Re-enable native cross-build

### PR DESCRIPTION
**Pull request by armanbilge**
A precursor for:
- https://github.com/typelevel/cats-effect/issues/1302

It seems the build was already mostly setup for native, this PR just re-enables it and freshens things up.

### CODE REVIEW

**Comment by armanbilge** [comment]
Posting this in case anyone with Native experience recognizes the issue.

```
specs2(topic/native)> coreNative/testOnly *.ApplicativeSpec
[info] Starting process '/workspace/specs2/common/native/target/scala-2.13/specs2-common-test-out' on port '33553'.
[info] Starting process '/workspace/specs2/core/native/target/scala-2.13/specs2-core-test-out' on port '32935'.
[info] Starting process '/workspace/specs2/core/native/target/scala-2.13/specs2-core-test-out' on port '45259'.
[warn] Force close java.lang.IllegalStateException: Unknown opcode: 4
[error] Test runner interrupted by fatal signal 9
[warn] Force close java.lang.RuntimeException: Process /workspace/specs2/core/native/target/scala-2.13/specs2-core-test-out finished with non-zero value 137
[error] stack trace is suppressed; run last coreNative / Test / testOnly for the full output
[error] (coreNative / Test / testOnly) scala.scalanative.testinterface.common.RPCCore$ClosedException: scala.scalanative.testinterface.NativeRunnerRPC$RunTerminatedException
[error] Total time: 3 s, completed Feb 15, 2022, 11:55:36 AM
```

**Comment by etorreborre** [comment]
Yes I haven't tried that yet (still working on a minimal specs2-cross version for both 2.13 and 3). You're brave!

**Comment by armanbilge** [comment]
That's great news! Fortunately @WojciechMazur was able to identify the problem but I haven't had a chance to push up the fix yet 😁

File: common/shared/src/main/scala/org/specs2/reflect/ClassName.scala [hunk 1]
```diff
@@ -65,7 +65,8 @@ trait ClassName { outer =>
       else              name
 
     }(klass.getName)
-    if (result.contains("anon") && klass.getSuperclass != null) simpleName(klass.getSuperclass)
+    if (Platform.isNative) result
+    else if (result.contains("anon") && klass.getSuperclass != null) simpleName(klass.getSuperclass)
     else result
```
Comment (armanbilge) [inline common/shared/src/main/scala/org/specs2/reflect/ClassName.scala:70]: The strange structure of the conditional is to help the Scala compiler remove the branch not linking on Native.

Linking to:
- https://github.com/scala-native/scala-native/issues/2560

File: core/shared/src/test/scala/org/specs2/specification/SharedInstantiationSpec.scala [hunk 2]
```diff
@@ -8,15 +8,15 @@ import org.specs2.main._
...
-class Shared1Spec extends mutable.Specification {
-  import scala.concurrent.ExecutionContext.Implicits.global
-  "This must work with the global Execution Context" should {
-    "match a future using both jvm + scalajs" in {
-      Future(ok)
-    }
-  }
-}
+// We don't use the global on JS (MacrotaskExecutor) or Native (ParasiticEC), so this spec has become irrelevant
+// class Shared1Spec extends mutable.Specification {
```
Comment (armanbilge) [inline core/shared/src/test/scala/org/specs2/specification/SharedInstantiationSpec.scala:12]: An unfortunate situation!

Comment (armanbilge) [reply core/shared/src/test/scala/org/specs2/specification/SharedInstantiationSpec.scala]: Ignore this, I found a better solution so we can keep this spec :)

**Comment by armanbilge** [comment]
@etorreborre that wasn't bad at all!

**Comment by etorreborre** [comment]
Wow, fantastic work! Next job for me: apply the same changes to specs2-5.0.0 on Scala 3 now that native is supported.

**Comment by etorreborre** [comment]
I'll try to release during the day or tomorrow because I'm travelling today.

**Comment by armanbilge** [comment]
Thanks! No rush on releasing, I did this primarily so that we can cross Cats Effect which will now be blocked by various other things 😅

Safe travels!

**Comment by etorreborre** [comment]
It's available as `4.14.0` for when you will be ready :-).

**Comment by armanbilge** [comment]
You're the best! 😁

**Pull request merged**